.travis.yml

```diff
@@ -1,3 +1,16 @@
-script:
-- docker build --target all_tests -t test . && docker run --name todo-app-test --env TRELLO_URL --env TRELLO_KEY --env TRELLO_BOARD_ID --env TRELLO_TOKEN test
-- docker build --target prod -t prod .
+jobs:
+  include:
+    - stage: test
+      script:
+        - docker build --target all_tests -t test .
+        - docker run --name todo-app-test --env TRELLO_URL --env TRELLO_KEY --env TRELLO_BOARD_ID --env TRELLO_TOKEN test
+    - stage: deploy
+      script:
+        - DOCKER_REPO="$DOCKER_USERNAME/todo-app"
+        - docker build --target prod -t "$DOCKER_REPO:$TRAVIS_COMMIT" -t "$DOCKER_REPO:latest" .
+        - docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+        - docker push $DOCKER_REPO
```

Comment:
FYI this relies on the fact that Travis does not use the latest version of Docker. With a recent Docker version you would need to either push the travis_commit image separately or include `--all-tags` here

Reply:
I think I originally had `--all-tags` but due to the old version it didn't recognise the flag